packages/@uppy/companion/src/server/provider/zoom/index.js

```diff
@@ -209,9 +211,15 @@ class Zoom extends Provider {
         modifiedDate: adapter.getDateFolderModified(end),
         size: null
       })
+
+      if (end.isSame(limit, 'month')) {
+        break
+      }
+
       end = start.clone().subtract(1, 'days')
     }
-    data.nextPagePath = adapter.getDateNextPagePath(start)
+
+    data.nextPagePath = adapter.getDateNextPagePath(start.clone(), allResultsShown)
```

Comment:
I think we can just do:

```js
data.nextPagePath = allResultsShown ? null : adapter.getDateNextPagePath(start.clone())
```

and leave `getDateNextPagePath` as it was before this PR

Reply:
makes sense, updated